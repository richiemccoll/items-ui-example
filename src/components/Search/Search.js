import React from "react";
import { PropTypes } from "prop-types";
import styled from 'styled-components';

const Search = styled.input`
    width: 80%;
    height: 2em;
    box-shadow: 0 6px 8px rgba(0, 0, 0, 0.15);
`;

class SearchBar extends React.Component {
  handleInput = e => {
    e.target.value && this.props.handleSearch(e.target.value);
  };

  render() {
    return <Search onChange={this.handleInput} placeholder="Search"/>;
  }
}

SearchBar.propTypes = {
  handleSearch: PropTypes.func.isRequired
};

export default SearchBar;
