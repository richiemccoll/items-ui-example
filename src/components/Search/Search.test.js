import React from "react";
import Search from "./Search";
import renderer from "react-test-renderer";

describe("Search Component", () => {
  it("should render correctly", () => {
    const component = renderer.create(<Search />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
