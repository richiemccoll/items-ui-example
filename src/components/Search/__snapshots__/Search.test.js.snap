// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render correctly 1`] = `
<input
  className="sc-bdVaJa dsiGYE"
  id="search"
  onChange={[Function]}
  placeholder="Search"
/>
`;
